cpp/rss_subset_aggregator.cc

```diff
@@ -0,0 +1,314 @@
+
+        const auto title(harvested_item.item_.getTitle());
+        if (not title.empty())
+            xml_writer->writeTagsWithData("title", harvested_item.item_.getTitle());
+
+        xml_writer->writeTagsWithData("link", harvested_item.item_.getLink());
+
+        const auto description(HtmlUtil::ShortenText(harvested_item.item_.getDescription(), /*max_length = */500));
+        if (not description.empty())
+            xml_writer->writeTagsWithData("description", description);
+
+        xml_writer->writeTagsWithData("pubDate",
+                                      TimeUtil::TimeTToString(harvested_item.item_.getPubDate(), TimeUtil::RFC822_FORMAT,
+                                                                     TimeUtil::UTC));
+        xml_writer->writeTagsWithData("guid", harvested_item.item_.getId());
+        xml_writer->closeTag("item", /* suppress_indent */ false);
+    }
+
+    xml_writer->closeTag("channel");
+    xml_writer->closeTag("rss");
+}
+
+
+struct FeedNameAndURL {
+    std::string name_;
+    std::string url_;
+public:
+    FeedNameAndURL() = default;
+    FeedNameAndURL(const FeedNameAndURL &other) = default;
+    FeedNameAndURL(const std::string &name, const std::string &url)
+        : name_(name), url_(url) { }
+};
+
+
+// \return True if a notification email was sent successfully, o/w false.
+bool SendEmail(const std::string &subsystem_type, const std::string &email_sender, const std::string &user_email,
+               const std::string &user_address, const std::string &language,
+               const std::vector<HarvestedRSSItem> &harvested_items)
+{
+    const auto template_filename_prefix(UBTools::GetTuelibPath() + "rss_email.template");
+    std::string template_filename(template_filename_prefix + "." + language);
+    if (not FileUtil::Exists(template_filename))
+        template_filename = template_filename_prefix + ".en";
+    static const std::string email_template(FileUtil::ReadStringOrDie(template_filename));
+
+    Template::Map names_to_values_map;
+    names_to_values_map.insertScalar("user_name", user_address);
+
+    std::vector<std::string> item_titles, item_urls, feed_urls, feed_names;
+    for (const auto &harvested_item : harvested_items) {
+        item_titles.emplace_back(HtmlUtil::HtmlEscape(harvested_item.item_.getTitle()));
+        item_urls.emplace_back(harvested_item.item_.getLink());
+        feed_urls.emplace_back(harvested_item.feed_url_);
+        feed_names.emplace_back(harvested_item.feed_title_);
+    }
+    names_to_values_map.insertArray("item_titles", item_titles);
+    names_to_values_map.insertArray("item_urls", item_urls);
+    names_to_values_map.insertArray("feed_urls", feed_urls);
+    names_to_values_map.insertArray("feed_names", feed_names);
+
+    const auto email_body(Template::ExpandTemplate(email_template, names_to_values_map));
+    const auto retcode(EmailSender::SimplerSendEmail(email_sender, { user_email }, GetChannelDescEntry(subsystem_type, "title"),
```

Comment:
The log is pretty empty so far... do you think it's a good thing to call LOG_INFO for every mail address where SentEmail was successful?

Reply:
No, not really.